crates/nu-parser/src/parser.rs

```diff
@@ -577,22 +577,23 @@ fn calculate_end_span(
             // keywords, they get to set this by being present
 
             let positionals_between = kw_pos - positional_idx - 1;
-            if positionals_between > (kw_idx - spans_idx) {
+            if positionals_between >= (kw_idx - spans_idx) {
```

Comment:
Why `>=` is required?

Reply:
I should have documented it when I wrote this change, as I don't remember. 😅
The commit this change comes from says:
>  **Handle multiple missing args before a keyword**
> calculate_end_span should only return spans_idx when the current
> span is a keyword, but we require more positional before the keyword

But I'll have to think a bit more to remember why that meant that we want `>=`. I should've also added another testcase for that specific change, since it's a separate fix from the previous ones.

Reply:
Yeah I have the question because I tried something like `let = 3` without this change, it also reports new error message.

Reply:
Yeah, I unfortunately still don't remember. I believe I had some reason for it, but it was a year ago and I didn't write enough down. I tried changing it to panic on `==` and running the fuzzer to see when the equals case would happen with no success. My commit message unfortunately wasn't clear enough for me to understand it, since I no longer remember how the parser works.

I would be fine with either reverting that change or keeping it. The test cases succeed in both cases, so it only makes a difference in some untested case (if ever).

The case of equality did trigger in a few tests, but for all of those, `positionals_between` was equal to 0, which means that the result of the two variants are identical.

It could be the case that the change only matters if there is a command with more than one required argument before a keyword argument, which there are currently none. But after trying out adding such a command, those seem to be broken enough to not even trigger this code path at all (I should ensure that there is no regression for that in this patch, even though no such command currently exist).

Reply:
Ah ok.  Thank you for the detailed information.  If you believe you had some reason for it, I think we can keep it.

> It could be the case that the change only matters if there is a command with more than one required argument before a keyword argument, which there are currently none. But after trying out adding such a command, those seem to be broken enough to not even trigger this code path at all (I should ensure that there is no regression for that in this patch, even though no such command currently exist).

Yeah I believe the parser had been changed a lot from the pr you created before.  So it might be checked and broken elsewhere.